src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -648,8 +648,10 @@ private PollingResult compareRemoteRevisionWithImpl(AbstractProject<?, ?> projec
             listener.getLogger().println("[poll] Last Built Revision: " + buildData.lastBuild.revision);
         }
 
-        final String singleBranch = "master";
-        // final String singleBranch = getSingleBranch(lastBuild);
+        final String singleBranch = getSingleBranch(lastBuild);
+        if (!singleBranch.equals("master")) {
+                return PollingResult.NO_CHANGES;
```

Comment:
Will this return no changes even if the build was triggered by a push to master?